docs/source-pytorch/common/trainer.rst

```diff
@@ -33,10 +33,20 @@ This is the basic use of the trainer:
 
 .. code-block:: python
```

Comment:
IMO this loses the initial purpose of the snippet which was to "show the basic use of the Trainer". The user now has to understand what is wikitext, random_split, a transformer etc

Reply:
My thought process here is that I can use Adrian's transformer and dataset as a consistent example throughout docs in order to replace the encoder-decoder as the base example.

Reply:
Replacing the encoder-decoder example is a good idea. But the previous code wasn't using that but skeletons that don't implement anything

Reply:
reverted in [426e8d5](https://github.com/Lightning-AI/lightning/pull/18452/commits/426e8d5c02e99798fbe473527d3e64023b245451)

Reply:
Note that none of the examples in this file use the encoder-decoder example. Will you revert them too?

Reply:
Yeah we need to put the import at the top here otherwise to the reader `LightningTransformer` is just a name without definition anywhere.

Reply:
@awaelchli there shouldn't be anymore uses of `LightningTransformer` as of [6671aa6](https://github.com/Lightning-AI/lightning/pull/18452/commits/6671aa6215041d1f8598854e39fd7a5c5330cb69)